nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -1124,6 +1124,26 @@ in
               https://docs.nextcloud.com/server/latest/admin_manual/configuration_database/db_conversion.html
             '';
           }
+          {
+            assertion =
+              lib.versionAtLeast overridePackage.version "32.0.0"
+              || (cfg.config.adminuser != null && cfg.config.adminpassFile != null);
+            message = ''
+              Disabling initial admin user creation is only available on Nextcloud >= 32.0.0.
+            '';
+          }
+          {
+            assertion = cfg.config.adminuser != null || cfg.config.adminpassFile == null;
+            message = ''
+              If `services.nextcloud.config.adminuser` is null, `services.nextcloud.config.adminpassFile` must be null as well in order to disable initial admin user creation.
+            '';
+          }
+          {
+            assertion = cfg.config.adminpassFile != null || cfg.config.adminuser == null;
```

Comment:
optional, but both assertions could be folded into a single assertion with something like `isNull adminpassFile == isNull adminuser`.

Reply:
Yes, but this way the error messages are a bit better. I can change it though, as it might be easier to read the code that way :woman_shrugging: